### PR TITLE
Cleanup plugin code

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import sbtprotoc.ProtocPlugin
+
+object AkkaGrpcPlugin extends AutoPlugin {
+  import sbtprotoc.ProtocPlugin.autoImport._
+
+  // don't enable automatically, you might not want to run it on every subproject automatically
+  override def trigger = noTrigger
+  override def requires = ProtocPlugin
+
+  trait Keys { _: autoImport.type =>
```
Comment (dwijnand) [inline sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala]: What's the advantage/point of doing things this way?

Comment (jrudolph) [reply sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala]: Do you mean the complete change or the weird self type?

Comment (dwijnand) [reply sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala]: the trait/self type setup

Comment (jrudolph) [reply sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala]: Anyway, I cannot remember why I've done it this way (the best PRs are the ones without any comment on why things were changed one way or the other), so if you feel like it we should probably change it to a less confusing variant.

Comment (dwijnand) [reply sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala]: Nope, just wondering if there were some learnings you'd gained that led you to this configuration.

**Pull request merged**